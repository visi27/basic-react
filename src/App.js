import React, { Component } from 'react'
import './App.css'

const list = [
  {
    title: 'React',
    url: 'https://facebook.github.io/react/',
    author: 'Jordan Walke',
    num_comments: 3,
    points: 4,
    objectID: 0,
  },
  {
    title: 'Redux',
    url: 'https://github.com/reactjs/redux',
    author: 'Dan Abramov, Andrew Clark',
    num_comments: 2,
    points: 5,
    objectID: 1,
  },
]

const isSearched = searchTerm => item =>
     item.title.toLowerCase().includes(searchTerm.toLowerCase())


class App extends Component {
  constructor (props) {
    super(props)
    this.state = {
      list,
      searchTerm: '',
    }
    this.onDismiss = this.onDismiss.bind(this)
    this.onAdd = this.onAdd.bind(this)
    this.onSearchChange = this.onSearchChange.bind(this)
    this.onReset = this.onReset.bind(this)
  }

  render () {
    return (
      <div className="App">
        <form>
          <input type="text" onChange={this.onSearchChange}/>
          <button type="button" onClick={this.onReset}>RESET</button>
        </form>
        {
          this.state.list.filter(isSearched(this.state.searchTerm)).map(item =>
            <div key={item.objectID}>
                <span>
                  <a href={item.url}>{item.title}</a>
                </span>
              <span>{item.author}</span>
              <span>{item.num_comments}</span>
              <span>{item.points}</span>
              <button
                type="button"
                onClick={() => this.onDismiss(item.objectID)}
              >
                Dismiss
              </button>
            </div>,
          )
        }
        <button type="button" onClick={this.onAdd}>ADD</button>
      </div>
    )
  }

  onDismiss (itemId) {
    this.setState({
      list: this.state.list.filter(item => {
        return item.objectID !== itemId
      }),
    })
  }

  onAdd () {
    const newItem = {
      title: 'Random',
      url: 'https://github.com/random/random',
      author: 'Random, Unknown',
      num_comments: Math.floor(Math.random() * 20),
      points: Math.floor(Math.random() * 10),
      objectID: Math.floor(Math.random() * 1000),
    }

    this.setState({
      list: [...this.state.list, newItem],
    })
  }

  onSearchChange (event) {
    this.setState({searchTerm: event.target.value})
  }

  onReset () {
    this.setState({
      list,
      searchTerm: '',
    })
  }
}

export default App
