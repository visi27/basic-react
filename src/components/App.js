import React, { Component } from "react";
import Search from "./Search/index";
import Table from "./Table/index";
import "./App.css";
import Button from "./Button/";
import withLoading from "../hoc/withLoading";

import { AppConfig as conf } from "../config";
import { fetchResults } from '../actions/result';
import { connect } from 'react-redux';

const ButtonWithLoading = withLoading(Button);

class App extends Component {
  componentDidMount() {
    this.props.onLoad();
  }

  render() {
    return (
      <div className="page">
        <span>HELLO</span>
        <div className="interactions">
          <Search> Search </Search>
        </div>

        <Table/>

        <ButtonWithLoading>
          More Articles
        </ButtonWithLoading>
      </div>
    );
  }
}

function mapDispatchToProps (dispatch) {
  return {
    onLoad: () => dispatch(fetchResults())
  }
}

export default connect(null, mapDispatchToProps)(App);
