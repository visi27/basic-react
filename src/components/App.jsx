import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Search from './Search/index';
import Table from './Table/index';
import './App.css';
import Button from './Button/presenter';
import withLoading from '../hoc/withLoading';
import fetchResults from '../actions/result';

const ButtonWithLoading = withLoading(Button);

export class App extends Component {
  componentDidMount() {
    this.props.onLoad();
  }

  render() {
    return (
      <div className="page">
        <span>HELLO</span>
        <div className="interactions">
          <Search> Search </Search>
        </div>

        <Table />

        <ButtonWithLoading>More Articles</ButtonWithLoading>
      </div>
    );
  }
}

App.propTypes = {
  onLoad: PropTypes.func.isRequired,
};

function mapDispatchToProps(dispatch) {
  return {
    onLoad: () => dispatch(fetchResults()),
  };
}

export default connect(null, mapDispatchToProps)(App);
