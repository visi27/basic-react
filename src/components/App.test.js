import React from 'react';
import ReactDOM from 'react-dom';
import { App } from './App';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('App', () => {
  test('has a valid snapshot', () => {
    const component = shallow(<App onLoad={()=>{}} />);
    let tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
