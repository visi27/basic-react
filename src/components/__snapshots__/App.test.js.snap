// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="page"
>
  <span>
    HELLO
  </span>
  <div
    className="interactions"
  >
    <Connect(Search)>
       Search 
    </Connect(Search)>
  </div>
  <Connect(Table) />
  <Component>
    More Articles
  </Component>
</div>
`;
